### PR TITLE
Update error message in multi_head_attention_forward, MultiheadAttention

### PR DESCRIPTION
**Pull request by vainaixr**

### CODE REVIEW

**Comment by vadimkantorov** [comment]
Fixes https://github.com/pytorch/pytorch/issues/28719

File: torch/nn/functional.py [hunk 1]
```diff
@@ -3764,7 +3764,7 @@ def multi_head_attention_forward(query,                           # type: Tensor
     assert key.size() == value.size()
 
     head_dim = embed_dim // num_heads
-    assert head_dim * num_heads == embed_dim, "embed_dim must be divisible by num_heads"
+    assert head_dim * num_heads == embed_dim, f"embed_dim = {embed_dim} must be divisible by num_heads = {num_heads}"
```
Comment (zou3519) [inline torch/nn/functional.py]: I don't think `f` strings are valid strings in Python 2, and we haven't fully dropped support for Python 2 yet (though that is coming soon: https://github.com/pytorch/pytorch/issues/23795)

**Review by vainaixr — COMMENTED** [review]
I update it

**Comment by zou3519** [comment]
I am not the best reviewer so I am requesting review from @zhangguanheng66

**Review by zhangguanheng66 — APPROVED** [review]
Thanks for the contribution. LGTM

File: torch/nn/modules/activation.py [hunk 2]
```diff
@@ -714,7 +714,7 @@ def __init__(self, embed_dim, num_heads, dropout=0., bias=True, add_bias_kv=Fals
         self.num_heads = num_heads
         self.dropout = dropout
         self.head_dim = embed_dim // num_heads
-        assert self.head_dim * num_heads == self.embed_dim, "embed_dim must be divisible by num_heads"
+        assert self.head_dim * num_heads == self.embed_dim, "embed_dim = {} must be divisible by num_heads = {}".format(embed_dim, num_heads)
```
Comment (zhangguanheng66) [inline torch/nn/modules/activation.py]: Need to fix the flake8 error.

**Review by zhangguanheng66 — CHANGES_REQUESTED** [review]

File: torch/nn/modules/activation.py (hunk 2 above)
Comment (vainaijr) [reply torch/nn/modules/activation.py]: should I change it to something like,
```
if self.head_dim * num_heads != self.embed_dim:
                    raise ValueError((
                       "embed_dim = {} must be divisible by num_heads = {}".format(embed_dim, num_heads)
```

Comment (zhangguanheng66) [reply torch/nn/modules/activation.py]: I think the line is just too long.

**Review by vainaixr — COMMENTED** [review]
does it give issue with tab use

File: torch/nn/modules/activation.py (hunk 2 above)
Comment (vainaijr) [reply torch/nn/modules/activation.py]: it still give error

Comment (zhangguanheng66) [reply torch/nn/modules/activation.py]: Can you point me the error? It seems the failed test case is irrelevant.

Comment (vainaijr) [reply torch/nn/modules/activation.py]: I meant it fails for one check caffe2 onnx ort2

Comment (zhangguanheng66) [reply torch/nn/modules/activation.py]: That one is from the upstream and it's irrelevant with your PR, IMO.

Comment (vainaijr) [reply torch/nn/modules/activation.py]: then it should be fine

**Comment by janosh** [comment]
Since pytorch is now 3.7+ better to use f-strings here.

**Pull request closed without merging**